static/js/chess.js

```diff
@@ -42,10 +53,97 @@ document.addEventListener('DOMContentLoaded', function() {
     function startNewGame() {
         gameState = initializeBoard();
         currentPlayer = 'white';
+        isGameOver = false;
         status.textContent = "White's Turn";
         updateBoard();
     }
 
+    function findKingPosition(color) {
+        const kingSymbol = color === 'white' ? '♔' : '♚';
+        for (const [position, piece] of gameState.pieces) {
+            if (piece.piece === kingSymbol) {
+                return position;
+            }
+        }
+        return null;
+    }
+
+    function isKingInCheck(color) {
+        const kingPos = findKingPosition(color);
+        if (!kingPos) return false;
+
+        const oppositeColor = color === 'white' ? 'black' : 'white';
+        for (const [position, piece] of gameState.pieces) {
+            if (piece.color === oppositeColor && isValidMove(position, kingPos, true)) {
+                return true;
+            }
+        }
+        return false;
```

Comment:
This should notify the player when the king is in check and which colour is in check